llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -18011,7 +18007,8 @@ static inline bool CanCombineFCOPYSIGN_EXTEND_ROUND(EVT XTy, EVT YTy) {
   if (YTy == MVT::f128)
     return false;
 
-  return !YTy.isVector() || EnableVectorFCopySignExtendRound;
+  // Avoid mismatched vector operand types, for better instruction selection.
```

Comment:
This was the original comment that I've pulled from https://reviews.llvm.org/D128642.